run.py

```diff
@@ -1,30 +1,77 @@
 import argparse
 
-from src.add_songs import create_db, add_track
+import config
+from src.ingest_data import run_ingest_data
+from src.clean_data import run_clean_data
+from src.generate_features import run_generate_features
+from src.create_db import run_create_db
 
-if __name__ == '__main__':
+if __name__ == "__main__":
 
-    # Add parsers for both creating a database and adding songs to it
-    parser = argparse.ArgumentParser(description="Create and/or add data to database")
+    # Add parsers for all functions in the pipeline
+    parser = argparse.ArgumentParser(description="Run model pipeline code")
     subparsers = parser.add_subparsers()
 
-    # Sub-parser for creating a database
-    sb_create = subparsers.add_parser("create_db", description="Create database")
-    sb_create.add_argument("--artist", default="Britney Spears", help="Artist of song to be added")
-    sb_create.add_argument("--title", default="Radar", help="Title of song to be added")
-    sb_create.add_argument("--album", default="Circus", help="Album of song being added.")
-    sb_create.add_argument("--engine_string", default='sqlite:///data/tracks.db',
-                           help="SQLAlchemy connection URI for database")
-    sb_create.set_defaults(func=create_db)
-
-    # Sub-parser for ingesting new data
-    sb_ingest = subparsers.add_parser("ingest", description="Add data to database")
-    sb_ingest.add_argument("--artist", default="Emancipator", help="Artist of song to be added")
-    sb_ingest.add_argument("--title", default="Minor Cause", help="Title of song to be added")
-    sb_ingest.add_argument("--album", default="Dusk to Dawn", help="Album of song being added")
-    sb_ingest.add_argument("--engine_string", default='sqlite:///data/tracks.db',
-                           help="SQLAlchemy connection URI for database")
-    sb_ingest.set_defaults(func=add_track)
+    # Sub-parser for ingesting data from source
+    sb_ingest = subparsers.add_parser(
+        "ingest", description="Ingest data from web source and upload raw file to S3."
+    )
+    sb_ingest.add_argument(
+        "--config", default=config.YAML_CONFIG, help="Location of configuration YAML"
+    )
+    sb_ingest.add_argument(
+        "--url", default=config.URL_LISTINGS, help="URL of source data"
+    )
+    sb_ingest.set_defaults(func=run_ingest_data)
+
+    # Sub-parser for cleaning data
+    sb_clean = subparsers.add_parser(
+        "clean", description="Get raw data from S3, clean data, and save as CSV.",
+    )
+    sb_clean.add_argument(
+        "--config", default=config.YAML_CONFIG, help="Location of configuration YAML"
```

Comment:
I am confused by the necessity of this. It would make the code more simple to access the config.YAML_CONFIG directly. I can see several scenarios where I can see someone editing some kind of configuration and there should be fewer: 
1/ I edit the YAML provided, which is called by config.YAML_CONFIG by config.py
2/I edit the actual path for config.py to another YAML
3/I change the location of the configuration YAML

I think you should remove the --config argument to reduce the amount of steps someone may take to change a configuration, but forget to change it elsewhere.

Reply:
Makes sense. Per your suggestion, I am removing the `--config` argument and instead just accessing the `config.YAML_CONFIG` directly within the modules.